modules/currency-l0/src/main/scala/io/constellationnetwork/currency/l0/StateChannel.scala

```diff
@@ -54,6 +55,11 @@ object StateChannel {
           case None => Applicative[F].unit
         }
 
+      def checkIfShouldForceEventTrigger(snapshot: GlobalIncrementalSnapshot, currencyId: Address): Boolean =
+        snapshot.spendActions.fold(false) { spendActions =>
```

Comment:
we're forcing the entire global snapshot sync but only querying into the `spendActions`?